avm/res/compute/gallery/application/main.bicep

```diff
@@ -124,28 +124,34 @@ output location string = application.location
 //   Definitions   //
 // =============== //
 
-type roleAssignmentType = {
-  @sys.description('Optional. The name (as GUID) of the role assignment. If not provided, a GUID will be generated.')
-  name: string?
+import { roleAssignmentType } from 'br/public:avm/utl/types/avm-common-types:0.3.0'
```

Comment:
While this is a valid place to import external UDTs, we agreed to import them at the location where they are used. This way it is clear where they come from.
I'd put the import roleAssignmentType right before the param roleAssignments. 
This applies to all imports and references of imported types (AVM common or your own).

Reply:
💡 Thanks for the feedback! Is the single line import required for this type as well?
`import { identifierType, purchasePlanType, resourceRangeType } from './image/main.bicep'`

Reply:
Nevermind, these are used within a UDT so must be in the one-liner :)

Reply:
One point may have gotten lost a bit. 

Hey @johnlokerse, long time no see. 😄 To build on top of what @ReneHezser said: Just for the types that you import from other templates, please import them on top of where they are used. For example:

```bicep
import { lockType } from 'br/public:avm/utl/types/avm-common-types:0.3.0'
@sys.description('Optional. The lock settings of the service.')
param lock lockType?
```

Reply:
Just makes it a bit easier to find - especially if the module would happen to not have any other types :) (in know that's not the case for you, but I wanted to mention it)

Reply:
Done! Also, I have moved the imports at the UDTs to above the `@export()`:

![CleanShot 2024-11-25 at 09 18 18@2x](https://github.com/user-attachments/assets/f53d0bf4-b4ca-43e7-9bda-4a4509f51bed)

Reply:
Great, thanks @johnlokerse. 🚀 
Just one ask - please check if you pushed your latest changes. In the PR I can see that e.g. the 'roleAssignmentType' is not on top of the parameter (yet others are).

Reply:
![Uploading image.png…]()